test/file.jl

```diff
@@ -1181,3 +1181,61 @@ if !Sys.iswindows()
         test_22566()
     end
 end  # !Sys.iswindows
+
+function test_22922()
+    def_prefix = "jl_"
+    tst_prefix = "ABCDEF"
+    mktempdir() do tmpdir
+        filename = basename(tmpdir)
+        @test startswith(filename, def_prefix)
+    end
+    mktempdir(; prefix=tst_prefix) do tmpdir
+        filename = basename(tmpdir)
+        @test startswith(filename, tst_prefix)
+    end
+    # Special character prefix tests
+    tst_prefix="#!@%^&()"
+    mktempdir(; prefix=tst_prefix) do tmpdir
+        filename = basename(tmpdir)
+        @test startswith(filename, tst_prefix)
+    end
+
+    # Behavioral differences across OS types
+    if Sys.iswindows()
+        @test_throws Base.UVError mktempdir(; prefix="*")
+        @test_throws Base.UVError mktempdir(; prefix="cdcdccd/")
+
+        # The API accepts "c:/ and c:\\" as valid prefixes.
+        # The parent directory is ignored in that case.
+
+        mkdir("c:/mydir")
+        # The temp directory created is of form "C:\\XXXXXX"
+        mktempdir("c:/mydir"; prefix="c:/") do tmpdir
+            filename = basename(tmpdir)
```

Comment:
Better also test the path before the base name to ensure that what the comment says actually happens. Same below and on Unix.

Reply:
Do you expect me to test for the following?
```
mktempdir("c:/mydir"; prefix="c:/") do tmpdir
   @test isdir(tmpdir)  
          # Does not makes much of sense as mktempdir() will create the dir() anyway. 
          # Our focus is mostly the name of the dir created. 
          # Those must be tested as part of regular mktempdir (without prefix) test case. 
   filename = basename(tmpdir)
```
The file already has a large number of `mktempdir(without prefix)` test cases.

Reply:
I meant that since you say "The temp directory created is of form "C:\\XXXXXX"", you should actually test that it's the case, i.e. that `startswith(tempdir, C:\\")`.